File: week1/pyramid.c
#include <cs50.h>
#include <stdio.h>


void draw_pyramid(int height);


int main(void)
{
    int height;
    do
    {
        // get input from user
        height = get_int("Height:\t");
    }
    while (height < 1 || height > 8);
    
    // draw a pyramid
    draw_pyramid(height);
}


// Draw a cell containing spaces or hashs
void draw_cell(int idx, int row)
{
    if (idx > row)
    {
        printf(" ");
    }
    else
    {
        printf("#");
    }
}


// draw a pyramid of a specific height
void draw_pyramid(int height)
{
    // draw rows
    for (int x = 1; x <= height; x++)
    {
        // left
        for (int i = height; i >= 1; i--)
        {
            draw_cell(i, x);
        }
        printf(" ");
        // right
        for (int i = 1; i <= height; i++)
        {
            draw_cell(i, x);
        }
        printf("\n");
    }
}

File: week6/pyramid.py
# from cs50 import get_int


def main():
    while True:
        n = input("Height: ")
        n = int(n)
        if n >= 1 and n < 9:
            break
    make_pyramid(n)

# Method to make a pyramid of height (n)


def make_pyramid(n):
    for i in range(n):
        # write the left half of the pyramid
        for j in reversed(range(n)):
            print(" " if j > i else "#", end="")
        
        # write two spaces
        print(" " * 2, end="")

        # write the right half of the pyramid
        for j in range(n):
            print("" if j > i else "#", end="")

        print()


main()
